### DISCUSSION TITLE
How to use non_max_suppression in Yolov8?

### DISCUSSION THREAD

**Discussion by M-Amrollahi** [ultralytics/ultralytics#5210] [Q&A]
When I see the results of object detection model by Yolov8, I can see several bboxes overlapped and I wonder if nms has been applied or not. How can I apply ```non_max_suppression``` to result of the model?
```python
results = model.predict(val_image)
```

**Comment by glenn-jocher** [comment]
@M-Amrollahi hello! Non-max suppression (NMS) is automatically applied in YOLOv8 when you use the `predict` mode. The `results` you get from `model.predict(val_image)` should already have NMS applied to filter out overlapping bounding boxes based on their confidence scores.

If you're seeing multiple overlapping boxes, it could be due to a lower confidence threshold or a higher IoU (Intersection over Union) threshold for NMS. You can adjust these parameters when calling `predict` to fine-tune the NMS process. Check out the documentation on the `predict` mode for more details on how to adjust these settings. 😊

If you need further assistance with tuning these parameters, feel free to ask!

**Reply by brahian204** [reply]
hola buenas tardes, por lo que he visto el algoritmo de nms se aplica para mascaras de segmentación, pero en mi caso lo necesito para los cuadros delimitadores, como podría integrar esa parte

**Reply by glenn-jocher** [reply]
@brahian204 ¡Hola! 😀 NMS se aplica automáticamente a los cuadros delimitadores (bounding boxes) en YOLOv8 cuando usas `predict`. Si necesitas ajustar el comportamiento de NMS, puedes especificar parámetros adicionales como el umbral de confianza (`conf_thres`) o el umbral IoU (`iou_thres`) al usar `predict`. Aquí tienes un ejemplo para modificar estos umbrales:

```python
results = model.predict(val_image, conf_thres=0.25, iou_thres=0.45)
```

Esto aplicará NMS con un umbral de confianza de 0.25 y un umbral IoU de 0.45. ¡Espero que esto te sea útil!
